File: unit01/w01_d04/homework/homework.js
### Problem 0:

var justDoIt = function(input) {
  output = input + ", JUST DO IT!";
  return output;
}

justDoIt("Will");


### Problem 1: Big or Small String?

var bigOrSmallString = function(string) {
  if (string.length > 10) {
    return "This word is big";
}  else {
    return "This string is small";
  }
};

bigOrSmallString("Poopy pants on fire");

### Problem 2: Odd or Even String Length?

var oddOrEvenString = function(string) {
  if (string.length % 2 === 0) {
    return "This string length is totally even";
}  else {
    return "This string's length is odd";
  }
};

oddOrEvenString("Poopy pants on fire!");

### Problem 3: Median

var returnMedianInArray = function(array) {
  sortedArray = array.sort(function(a, b){return a-b});
  // median = arethmic mean of the middle values
  var middle = Math.floor(sortedArray.length/2);
  if(sortedArray.length % 2) {
    median = (sortedArray[middle]);
 } else {
    median = ((sortedArray[middle-1] + sortedArray[middle]) / 2);
 }
 return median;
};

var nums = [14,11,16,15,13,16,15,17,19,11,12,14,19,11,15,17,11,18,12,17,12,71,18,15,12];
returnMedianInArray(nums);

### Problem 4: Positive or Negative?

var posNeg = function(int1,int2) {
  if (int1 > 0 && int2 < 0) {
    return true;
} else if (int1 < 0 && int2 > 0) {
  return true;
 } else {
    return false;
 }
};

posNeg(1, -1)

### Problem 5: Double X Counter

var countXx = function(string) {
  var count = 0;
  for (i=0; i<string.length; i++) {
    if (string[i] === "x" && string[i+1] === "x") {
      count++;
    }
  }
    return count;
};
  //countXx("abcxx");
  // 1
//  countXx("xxx");
  // 2
//  countXx("xxxx");
  // 3

### Problem 6: Initials

var initials = function(string) {
  initials = "";
  names = string.split(" ");
  for (i=0; i<names.length; i++) {
    currentName = names[i];
    initials = initials + currentName[0];
  }
  return initials;
};

initials('Neil Patrick Harris');
// NPH

### Problem 7: twoLengths

var twoLengths = function(string1,string2) {
  lengthArray = [];
  lengthArray[0] = (string1.length);
  lengthArray[1] = (string2.length);
  return lengthArray;
};
twoLengths("Hank", "Hippopopalous");
// => [4, 13]

## Problem 8: Reverse

var wordReverse = function(string) {
  splitString = string.split(" ");
  reversedString = "";
  for (i=0; i<splitString.length; i++) {
    console.log(splitString[i]);
    reversedString = splitString[i] + " " + reversedString;
    console.log(reversedString);
  }
  return reversedString.trim();
};

wordReverse("Ishmael me Call")
// => "Call me Ishmael"

##Function Break! Continuation of GA Press Release
##In Joe's lecture today, you were working on a General Assembly Press Release. Since we are all about reps, please finish that exercise if you did not complete it in class today.

## Bonus Stage:

## Problem 9: Longest string

var longest = function(array) {
  longest = "";
  for (i=0; i < array.length; i++) {
    if (array[i].length > longest.length) {
      longest = array[i];
    }
  }
  return longest;
};

// example usage
longest(["BoJack", "Princess", "Diane", "a", "Peanutbutter", "big", "Todd"]);
// => "Peanutbutter"

## Problem 10: Daffy

var toonify = function(accent, sentence) {
  if (accent != "daffy") {
  return sentence;
} else {
  newSentence = "";
    for (i=0; i < sentence.length; i++) {
      if (sentence[i] === "s") {
        newSentence = newSentence + "th";
      } else {
        newSentence = newSentence + sentence[i];
      }
    }
  }
  return newSentence;
};
toonify("daffy", "so you smell like sausage")
// => "tho you thmell like thauthage"

## Problem 11: Test Prime

var testPrime = function(number) {
  result = false;
  start = Math.ceil(number / 2);
    console.log(start);
  for (i=start; i > 1; i--) {
    console.log(i);
    console.log(number % i);
    if (number % i === 0) {
      return false;
  } else {
        return true;
    }
  }
};

testPrime(37);
// => true;

#### Finished? :trophy:

File: unit01/w01_d04/homework/homework.rb
Problem 0: Just DO it!!

Write a function justDoIt that accepts one parameter an argument. The function should return a string that has your argument and the phrase "JUST DO IT".

def justDoIt(argument)
  newString = argument + ", JUST DO IT"
  puts newString
  return newString
end
justDoIt("Will");

Problem 1: Big or Small String?

def bigOrSmallString(string)
  if string.length > 10
    return "This word is big"
  else
    return "This string is small"
  end
end
bigOrSmallString("Will");

Problem 2: Odd or Even String Length?

def oddOrEvenString(string)
  if string.length % 2 === 0
    return "This string length is totally even"
  else
    return "This string's length is odd"
  end
end
oddOrEvenString("Will");

Problem 3: Median

def findMedianInArray(nums)
  sortedArray = nums.sort
  middle = (sortedArray.length/2).floor
  if(sortedArray.length % 2)
    median = (sortedArray[middle])
  else
    median = ((sortedArray[middle-1] + sortedArray[middle]) / 2)
  end
  return median
end
nums = [14,11,16,15,13,16,15,17,19,11,12,14,19,11,15,17,11,18,12,17,12,71,18,15,12]
findMedianInArray(nums)

Problem 4: Positive or Negative?

def posNeg(num1,num2)
  if num1 > 0 && num2 < 0
    return true
  elsif num1 < 0 && num2 > 0
    return true
  else
    return false
  end
end

#posNeg(1, -1)
# true
#posNeg(-55, 1)
# true
#posNeg(-4, -5)
# false

Problem 5: Double X Counter

def countXx(string)
  count = 0
  i = 0
  while i < string.length do
    if string[i] === "x" && string[i+1] === "x"
      count = count + 1
    end
    i +=1
  end
  return count
end

#countXx("abcxx")
#1
#countXx("xxx")
#2
#countXx("xxxx")
#3

Problem 6: Initials

Write a function called initials that accepts one argument, a string. Given a person's name, return their initials.

It should work both with and without a middle name being provided.
initials('Neil Patrick Harris');
// NPH
Problem 7: twoLengths

Write a function twoLengths that accepts two parameters as an argument. The function should return an array of numbers where each number is the length of the corresponding string.

// example usage
twoLengths("Hank", "Hippopopalous");
// => [4, 13]
Problem 8: Reverse

Write a function wordReverse that accepts a single argument, a string. The method should return a string with the order of the words reversed. Don't worry about punctuation.

// example usage
wordReverse("Ishmael me Call")
// => "Call me Ishmael"
Function Break! Continuation of GA Press Release

In Joe's lecture today, you were working on a General Assembly Press Release. Since we are all about reps, please finish that exercise if you did not complete it in class today.

Bonus Stage:

Problem 9: Longest string

Write a function longest that accepts a single argument, an array of strings. The method should return the longest word in the array. In case of a tie, the method should return the word that appears first in the array.

// example usage
longest(["BoJack", "Princess", "Diane", "a", "Peanutbutter", "big", "Todd"]);
// => "Peanutbutter"
Problem 10: Daffy

Write a function called toonify that takes two parameters, accent and sentence.

If accent is the string "daffy", return a modified version of sentence with all "s" replaced with "th".
If the accent is not daffy, just return the sentence as-is.
// example usage
toonify("daffy", "so you smell like sausage")
// => "tho you thmell like thauthage"
Problem 11: Test Prime

Write a function testPrime that accepts a number as a parameter and check the number is prime or not.

Hint : A prime number (or a prime) is a natural number greater than 1 that has no positive divisors other than 1 and itself.

testPrime(37);
// => true;
